servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -3482,6 +3482,8 @@ public void putMembership(ResourceContext ctx, String domainName, String roleNam
 
         verifyAuthorizedServiceRoleOperation(principal.getAuthorizedService(), caller, roleName);
 
+        validateNotMtlsRestricted(ctx, caller);
```

Comment:
For ZMS - why not just put the check in the authorize call. We should not allow any write operations with the restricted cert.